pkg/controller/garbagecollector/metrics.go

```diff
@@ -65,9 +75,47 @@ func Register() {
 		prometheus.MustRegister(EventProcessingLatency)
 		prometheus.MustRegister(DirtyProcessingLatency)
 		prometheus.MustRegister(OrphanProcessingLatency)
+		prometheus.MustRegister(DeletionToWatchLatency)
 	})
 }
 
 func sinceInMicroseconds(clock clock.Clock, start time.Time) float64 {
 	return float64(clock.Since(start).Nanoseconds() / time.Microsecond.Nanoseconds())
 }
+
+func NewDeletionToWatchTracker(limit int) *uidToDeletionTime {
+	return &uidToDeletionTime{
+		items: make(map[types.UID]time.Time),
+		lock:  sync.Mutex{},
+		limit: limit,
+	}
+}
+
+type uidToDeletionTime struct {
+	items map[types.UID]time.Time
+	lock  sync.Mutex
+	limit int
+}
+
+// DeletionSent should be called when the deletion request is sent.
+func (u *uidToDeletionTime) DeletionSent(uid types.UID, start time.Time) error {
+	u.lock.Lock()
+	defer u.lock.Unlock()
+	if len(u.items) >= u.limit {
+		return fmt.Errorf("limit %d is reached", u.limit)
+	}
+	u.items[uid] = start
```

Comment:
Do you want to do anything special if it's already in there, like if we send two delete requests?